Unresolvable 'new' Indentation Rule conflict introduced in 8.38

The following code snippet exists in my codebase and passes Checkstyle using 8.37 rules.  When run against the 8.38 rules, I get violations due to updated indentation rules in the release: "'operator new' has incorrect indentation level 24, expected level should be one of the following: 28, 32."

```
                redirectAttributes.addFlashAttribute("globalMessage",
                        new GenericResponse(ResponseStatus.ERROR, messages.getMessage("data.trigger.enable.failure",
                        new Object[] { ERROR_MSG })));
```

I changed the indentation level to 28, and got this Checkstyle rule violation:
"'new' has incorrect indentation level 28, expected level should be 24."
```
                redirectAttributes.addFlashAttribute("globalMessage",
                        new GenericResponse(ResponseStatus.ERROR, messages.getMessage("data.trigger.enable.failure",
                            new Object[] { ERROR_MSG })));
```

I also tried changing the indentation level to 32, and got this Checkstyle rule violation:
"'new' has incorrect indentation level 32, expected level should be 24."
```
                redirectAttributes.addFlashAttribute("globalMessage",
                        new GenericResponse(ResponseStatus.ERROR, messages.getMessage("data.trigger.enable.failure",
                                new Object[] { ERROR_MSG })));
```


[checkstyle.xml.zip](https://github.com/checkstyle/checkstyle/files/5622635/checkstyle.xml.zip)

Test.java
```
class T {
    public String test() {
        redirectAttributes.addFlashAttribute("globalMessage",
                new GenericResponse(ResponseStatus.ERROR, messages.getMessage("data.trigger.enable.failure",
                        new Object[] { ERROR_MSG })));
    }
}
```

config:
```
<module name="Indentation">
      <property name="basicOffset" value="4"/>
      <property name="braceAdjustment" value="0"/>
      <property name="caseIndent" value="4"/>
      <property name="throwsIndent" value="4"/>
      <property name="lineWrappingIndentation" value="8"/>
      <property name="forceStrictCondition" value="true" />
</module>
```

# checkstyle 8.38: 
```
Starting audit...
[ERROR] /home/abhishhek/playground/Test.java:5:25: 'new' has incorrect indentation level 24, expected level should be 16. [Indentation]
Audit done.
Checkstyle ends with 1 errors.
```
when indented on 16, it generates new error:
```
Starting audit...
[ERROR] /home/abhishhek/playground/Test.java:5:17: 'operator new' has incorrect indentation level 16, expected level should be one of the following: 20, 24. [Indentation]
Audit done.
Checkstyle ends with 1 errors.
```
# Checkstyle 8.37 and below:
```
Starting audit...
[ERROR] /home/abhishhek/playground/Test.java:5:25: 'new' has incorrect indentation level 24, expected level should be 16. [Indentation]
Audit done.
Checkstyle ends with 1 errors.
```
when indented on 16, everything passes: (which is desired).

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/LineWrappingHandler.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/LineWrappingHandler.java
@@ -70,6 +70,21 @@ public class LineWrappingHandler {
     }
 
     /**
+     * The list of ignored token types for being checked by lineWrapping indentation
+     * inside {@code checkIndentation()} as these tokens are checked for lineWrapping
+     * inside their dedicated handlers.
+     *
+     * @see NewHandler#getIndentImpl()
+     * @see BlockParentHandler#curlyIndent()
+     * @see ArrayInitHandler#getIndentImpl()
+     */
+    private static final int[] IGNORED_LIST = {
+        TokenTypes.RCURLY,
+        TokenTypes.LITERAL_NEW,
+        TokenTypes.ARRAY_INIT,
+    };
+
+    /**
      * The current instance of {@code IndentationCheck} class using this
      * handler. This field used to get access to private fields of
      * IndentationCheck instance.
@@ -162,7 +177,7 @@ public class LineWrappingHandler {
             if (currentType == TokenTypes.RPAREN) {
                 logWarningMessage(node, firstNodeIndent);
             }
-            else if (currentType != TokenTypes.RCURLY && currentType != TokenTypes.ARRAY_INIT) {
+            else if (!TokenUtil.isOfType(currentType, IGNORED_LIST)) {
                 logWarningMessage(node, currentIndent);
             }
         }

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/NewHandler.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/NewHandler.java
@@ -42,19 +42,20 @@ public class NewHandler extends AbstractExpressionHandler {
     public NewHandler(IndentationCheck indentCheck,
                       DetailAST ast,
                       AbstractExpressionHandler parent) {
-        super(indentCheck, "operator new", ast, parent);
+        super(indentCheck, "new", ast, parent);
         mainAst = ast;
     }
 
     @Override
     public void checkIndentation() {
         // if new is on the line start and it is not the part of assignment.
-        if (isOnStartOfLine(mainAst)
-                && !isNewKeywordFollowedByAssign()) {
+        if (isOnStartOfLine(mainAst)) {
             final int columnNo = expandedTabsColumnNo(mainAst);
             final IndentLevel level = getIndentImpl();
 
-            if (columnNo < level.getFirstIndentLevel()) {
+            final boolean forceStrictCondition = getIndentCheck().isForceStrictCondition();
+            if (forceStrictCondition && !level.isAcceptable(columnNo)
+                || !forceStrictCondition && level.isGreaterThan(columnNo)) {
                 logError(mainAst, "", columnNo, level);
             }
         }
@@ -88,7 +89,10 @@ public class NewHandler extends AbstractExpressionHandler {
         if (getLineStart(mainAst) == mainAst.getColumnNo()) {
             result = super.getIndentImpl();
 
-            if (isNewKeywordFollowedByAssign()) {
+            final boolean isLineWrappedNew = TokenUtil.isOfType(mainAst.getParent().getParent(),
+                                        TokenTypes.ASSIGN, TokenTypes.LITERAL_RETURN);
+
+            if (isLineWrappedNew || doesChainedMethodNeedsLineWrapping()) {
                 result = new IndentLevel(result, getLineWrappingIndent());
             }
         }
@@ -97,15 +101,6 @@ public class NewHandler extends AbstractExpressionHandler {
         }
 
         return result;
-    }
-
-    /**
-     * Checks if the 'new' keyword is followed by an assignment.
-     *
-     * @return true if new keyword is followed by assignment.
-     */
-    private boolean isNewKeywordFollowedByAssign() {
-        return mainAst.getParent().getParent().getType() == TokenTypes.ASSIGN;
     }
 
     /**
@@ -121,6 +116,23 @@ public class NewHandler extends AbstractExpressionHandler {
     @Override
     protected boolean shouldIncreaseIndent() {
         return false;
+    }
+
+    /**
+     * The function checks if the new keyword is a child of chained method calls,
+     * it checks if the new is directly followed by equal operator or return operator.
+     *
+     * @return true if the new it is chained method calls and new keyword is directly followed
+     *         by assign or return
+     */
+    private boolean doesChainedMethodNeedsLineWrapping() {
+        DetailAST ast = mainAst.getParent();
+
+        while (TokenUtil.isOfType(ast, TokenTypes.DOT, TokenTypes.METHOD_CALL, TokenTypes.EXPR)) {
+            ast = ast.getParent();
+        }
+
+        return TokenUtil.isOfType(ast, TokenTypes.ASSIGN, TokenTypes.LITERAL_RETURN);
     }
 
 }

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/indentation/IndentationCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/indentation/IndentationCheckTest.java
@@ -389,12 +389,12 @@ public class IndentationCheckTest extends AbstractModuleTestSupport {
         checkConfig.addAttribute("tabWidth", "4");
         checkConfig.addAttribute("throwsIndent", "4");
         final String[] expected = {
-            "33:9: " + getCheckMessage(MSG_ERROR_MULTI, "operator new", 8, "10, 12"),
-            "40:9: " + getCheckMessage(MSG_ERROR_MULTI, "operator new", 8, "10, 12"),
-            "90:11: " + getCheckMessage(MSG_ERROR_MULTI, "operator new", 10, "12, 14"),
-            "97:13: " + getCheckMessage(MSG_ERROR_MULTI, "operator new", 12, "14, 16"),
-            "119:13: " + getCheckMessage(MSG_ERROR_MULTI, "operator new", 12, "14, 16"),
-            "126:15: " + getCheckMessage(MSG_ERROR_MULTI, "operator new", 14, "16, 18"),
+            "33:9: " + getCheckMessage(MSG_ERROR_MULTI, "new", 8, "10, 12"),
+            "40:9: " + getCheckMessage(MSG_ERROR_MULTI, "new", 8, "10, 12"),
+            "90:11: " + getCheckMessage(MSG_ERROR_MULTI, "new", 10, "12, 14"),
+            "97:13: " + getCheckMessage(MSG_ERROR_MULTI, "new", 12, "14, 16"),
+            "119:13: " + getCheckMessage(MSG_ERROR_MULTI, "new", 12, "14, 16"),
+            "126:15: " + getCheckMessage(MSG_ERROR_MULTI, "new", 14, "16, 18"),
         };
         verifyWarns(checkConfig, getPath("InputIndentationCorrectIfAndParameter.java"), expected);
     }
@@ -811,9 +811,9 @@ public class IndentationCheckTest extends AbstractModuleTestSupport {
             "100:11: " + getCheckMessage(MSG_CHILD_ERROR, "if", 10, 12),
             "101:7: " + getCheckMessage(MSG_ERROR, "if rcurly", 6, 8),
             "104:11: " + getCheckMessage(MSG_ERROR, "Arrays", 10, 12),
-            "110:15: " + getCheckMessage(MSG_ERROR, "operator new", 14, 16),
+            "110:15: " + getCheckMessage(MSG_ERROR, "new", 14, 16),
             "113:11: " + getCheckMessage(MSG_CHILD_ERROR, "method call", 10, 12),
-            "118:15: " + getCheckMessage(MSG_ERROR, "operator new", 14, 16),
+            "118:15: " + getCheckMessage(MSG_ERROR, "new", 14, 16),
             "122:11: " + getCheckMessage(MSG_ERROR, "new", 10, 12),
             "126:11: " + getCheckMessage(MSG_ERROR, "new", 10, 12),
             "127:7: " + getCheckMessage(MSG_ERROR, ")", 6, 8),
@@ -1031,16 +1031,47 @@ public class IndentationCheckTest extends AbstractModuleTestSupport {
     }
 
     @Test
-    public void testNewChildren() throws Exception {
+    public void testNewKeyword2() throws Exception {
         final DefaultConfiguration checkConfig = createModuleConfig(IndentationCheck.class);
 
         checkConfig.addAttribute("basicOffset", "4");
-        checkConfig.addAttribute("forceStrictCondition", "false");
+        checkConfig.addAttribute("forceStrictCondition", "true");
         checkConfig.addAttribute("lineWrappingIndentation", "8");
         checkConfig.addAttribute("tabWidth", "4");
         checkConfig.addAttribute("throwsIndent", "8");
         final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;
-        verifyWarns(checkConfig, getPath("InputIndentationValidNewChildren.java"), expected);
+        verifyWarns(checkConfig, getPath("InputIndentationNew.java"), expected);
+    }
+
+    @Test
+    public void testValidNewKeywordWithForceStrictCondition() throws Exception {
+        final DefaultConfiguration checkConfig = createModuleConfig(IndentationCheck.class);
+
+        checkConfig.addAttribute("basicOffset", "4");
+        checkConfig.addAttribute("forceStrictCondition", "true");
+        checkConfig.addAttribute("lineWrappingIndentation", "8");
+        checkConfig.addAttribute("tabWidth", "4");
+        checkConfig.addAttribute("throwsIndent", "8");
+        final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;
+        verifyWarns(checkConfig, getPath("InputIndentationNew.java"), expected);
+    }
+
+    @Test
+    public void testInvalidNewKeywordWithForceStrictCondition() throws Exception {
+        final DefaultConfiguration checkConfig = createModuleConfig(IndentationCheck.class);
+
+        checkConfig.addAttribute("basicOffset", "4");
+        checkConfig.addAttribute("forceStrictCondition", "true");
+        checkConfig.addAttribute("lineWrappingIndentation", "8");
+        checkConfig.addAttribute("tabWidth", "4");
+        checkConfig.addAttribute("throwsIndent", "8");
+        final String[] expected = {
+            "23:17: " + getCheckMessage(MSG_ERROR, "new", 16, 24),
+            "24:21: " + getCheckMessage(MSG_CHILD_ERROR_MULTI, "object def", 20, "28, 32, 36"),
+            "25:17: " + getCheckMessage(MSG_ERROR_MULTI, "object def rcurly", 16, "24, 28, 32"),
+        };
+        verifyWarns(checkConfig,
+            getPath("InputIndentationNewWithForceStrictCondition.java"), expected);
     }
 
     @Test
@@ -1110,6 +1141,93 @@ public class IndentationCheckTest extends AbstractModuleTestSupport {
                 "8, 31, 33"),
             "53:5: " + getCheckMessage(MSG_CHILD_ERROR_MULTI, "array initialization",
                     4, "8, 31, 33"),
+            "58:7: " + getCheckMessage(MSG_CHILD_ERROR, "array initialization", 6, 8),
+            "63:3: " + getCheckMessage(MSG_ERROR, "member def type", 2, 4),
+            "65:7: " + getCheckMessage(MSG_ERROR, "member def type", 6, 4),
+            "66:3: " + getCheckMessage(MSG_ERROR_MULTI, "array initialization rcurly", 2, "6, 10"),
+            "69:7: " + getCheckMessage(MSG_CHILD_ERROR, "array initialization", 6, 8),
+            "76:11: " + getCheckMessage(MSG_CHILD_ERROR, "array initialization", 10, 12),
+            "89:9: " + getCheckMessage(MSG_ERROR, "1", 8, 12),
+            "100:11: " + getCheckMessage(MSG_CHILD_ERROR, "array initialization", 10, 12),
+            "101:15: " + getCheckMessage(MSG_CHILD_ERROR, "array initialization", 14, 12),
+            "104:11: " + getCheckMessage(MSG_CHILD_ERROR, "array initialization", 10, 12),
+            "105:15: " + getCheckMessage(MSG_CHILD_ERROR, "array initialization", 14, 12),
+            "106:7: " + getCheckMessage(MSG_ERROR_MULTI, "array initialization rcurly", 6, "8, 12"),
+            "109:7: " + getCheckMessage(MSG_ERROR_MULTI, "array initialization lcurly", 6, "8, 12"),
+            "110:15: " + getCheckMessage(MSG_CHILD_ERROR, "array initialization", 14, 12),
+            "111:11: " + getCheckMessage(MSG_CHILD_ERROR, "array initialization", 10, 12),
+            "112:7: " + getCheckMessage(MSG_ERROR_MULTI, "array initialization rcurly", 6, "8, 12"),
+            // following are tests for annotation array initialisation
+            "120:13: " + getCheckMessage(MSG_CHILD_ERROR_MULTI, "annotation array initialization",
+                12, "16, 46, 48"),
+            "124:15: " + getCheckMessage(MSG_CHILD_ERROR, "annotation array initialization",
+                14, 12),
+            "128:15: " + getCheckMessage(MSG_CHILD_ERROR_MULTI, "annotation array initialization",
+                14, "16, 28, 30"),
+            "129:9: " + getCheckMessage(MSG_ERROR_MULTI, "annotation array initialization rcurly",
+                8, "12, 16"),
+            "131:13: " + getCheckMessage(MSG_CHILD_ERROR, "annotation array initialization",
+                12, 16),
+        };
+
+        // Test input for this test case is not checked due to issue #693.
+        verify(checkConfig, fileName, expected);
+    }
+
+    @Test
+    public void testChainedMethodCalling() throws Exception {
+        final DefaultConfiguration checkConfig = createModuleConfig(IndentationCheck.class);
+
+        checkConfig.addAttribute("arrayInitIndent", "2");
+        checkConfig.addAttribute("basicOffset", "2");
+        checkConfig.addAttribute("braceAdjustment", "2");
+        checkConfig.addAttribute("caseIndent", "2");
+        checkConfig.addAttribute("forceStrictCondition", "false");
+        checkConfig.addAttribute("lineWrappingIndentation", "4");
+        checkConfig.addAttribute("tabWidth", "4");
+        checkConfig.addAttribute("throwsIndent", "4");
+        final String fileName = getPath("InputIndentationChainedMethodCalls.java");
+        final String[] expected = {
+            "32:5: " + getCheckMessage(IndentationCheck.class,
+                MSG_CHILD_ERROR, "method call", 4, 8),
+            "37:5: " + getCheckMessage(IndentationCheck.class, MSG_ERROR, ".", 4, 8),
+            "38:5: " + getCheckMessage(IndentationCheck.class, MSG_ERROR, ".", 4, 8),
+            "41:5: " + getCheckMessage(IndentationCheck.class, MSG_ERROR, "new", 4, 8),
+        };
+        verifyWarns(checkConfig, fileName, expected);
+    }
+
+    @Test
+    public void testInvalidArrayInitWithTrueStrictCondition()
+            throws Exception {
+        final DefaultConfiguration checkConfig = createModuleConfig(IndentationCheck.class);
+
+        checkConfig.addAttribute("arrayInitIndent", "4");
+        checkConfig.addAttribute("basicOffset", "4");
+        checkConfig.addAttribute("braceAdjustment", "0");
+        checkConfig.addAttribute("caseIndent", "4");
+        checkConfig.addAttribute("forceStrictCondition", "true");
+        checkConfig.addAttribute("lineWrappingIndentation", "4");
+        checkConfig.addAttribute("tabWidth", "4");
+        checkConfig.addAttribute("throwsIndent", "4");
+        final String fileName = getPath("InputIndentationInvalidArrayInitIndent.java");
+        final String[] expected = {
+            "21:3: " + getCheckMessage(MSG_ERROR, "member def type", 2, 4),
+            "22:7: " + getCheckMessage(MSG_ERROR, "member def type", 6, 4),
+            "24:3: " + getCheckMessage(MSG_ERROR, "member def type", 2, 4),
+            "28:7: " + getCheckMessage(MSG_ERROR, "member def type", 6, 4),
+            "29:9: " + getCheckMessage(MSG_CHILD_ERROR, "array initialization", 8, 10),
+            "30:5: " + getCheckMessage(MSG_ERROR_MULTI, "array initialization rcurly", 4, "6, 10"),
+            "33:10: " + getCheckMessage(MSG_CHILD_ERROR, "array initialization", 9, 8),
+            "34:8: " + getCheckMessage(MSG_CHILD_ERROR, "array initialization", 7, 8),
+            "35:10: " + getCheckMessage(MSG_CHILD_ERROR, "array initialization", 9, 8),
+            "40:3: " + getCheckMessage(MSG_ERROR_MULTI, "array initialization lcurly", 2, "4, 8"),
+            "44:7: " + getCheckMessage(MSG_ERROR_MULTI, "array initialization rcurly", 6, "4, 8"),
+            "48:3: " + getCheckMessage(MSG_ERROR_MULTI, "array initialization lcurly", 2, "4, 8"),
+            "52:21: " + getCheckMessage(MSG_CHILD_ERROR_MULTI, "array initialization", 20,
+                "8, 31, 33"),
+            "53:5: " + getCheckMessage(MSG_CHILD_ERROR_MULTI, "array initialization",
+                4, "8, 31, 33"),
             "58:7: " + getCheckMessage(MSG_CHILD_ERROR, "array initialization", 6, 8),
             "63:3: " + getCheckMessage(MSG_ERROR, "member def type", 2, 4),
             "65:7: " + getCheckMessage(MSG_ERROR, "member def type", 6, 4),
@@ -2314,6 +2432,7 @@ public class IndentationCheckTest extends AbstractModuleTestSupport {
             "60:8: " + getCheckMessage(MSG_ERROR, "annotation field def modifier", 7, 4),
             "61:5: " + getCheckMessage(MSG_ERROR, "annotation def rcurly", 4, 0),
             "72:4: " + getCheckMessage(MSG_ERROR, "annotation def modifier", 3, 4),
+            "87:29: " + getCheckMessage(MSG_ERROR_MULTI, "new", 28, "20, 24"),
             "117:6: " + getCheckMessage(MSG_ERROR, "annotation def modifier", 5, 4),
             "128:2: " + getCheckMessage(MSG_ERROR, "interface", 1, 0),
             "134:12: " + getCheckMessage(MSG_ERROR, "@", 11, 0),
@@ -2435,8 +2554,21 @@ public class IndentationCheckTest extends AbstractModuleTestSupport {
         final String[] expected = {
             "8:1: " + getCheckMessage(MSG_ERROR, "Object", 0, 12),
             "10:1: " + getCheckMessage(MSG_ERROR, "(", 0, 12),
-            "13:1: " + getCheckMessage(MSG_CHILD_ERROR, "operator new", 0, 8),
-            "15:1: " + getCheckMessage(MSG_ERROR, "operator new lparen", 0, 8),
+            "13:1: " + getCheckMessage(MSG_CHILD_ERROR, "new", 0, 8),
+            "15:1: " + getCheckMessage(MSG_ERROR, "new lparen", 0, 8),
+        };
+        verifyWarns(checkConfig, getPath("InputIndentationNewHandler.java"), expected);
+    }
+
+    @Test
+    public void testNewHandler2() throws Exception {
+        final DefaultConfiguration checkConfig = createModuleConfig(IndentationCheck.class);
+        checkConfig.addAttribute("tabWidth", "4");
+        final String[] expected = {
+            "8:1: " + getCheckMessage(MSG_ERROR, "Object", 0, 12),
+            "10:1: " + getCheckMessage(MSG_ERROR, "(", 0, 12),
+            "13:1: " + getCheckMessage(MSG_CHILD_ERROR, "new", 0, 8),
+            "15:1: " + getCheckMessage(MSG_ERROR, "new lparen", 0, 8),
         };
         verifyWarns(checkConfig, getPath("InputIndentationNewHandler.java"), expected);
     }

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationChainedMethodCalls.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationChainedMethodCalls.java
@@ -1,0 +1,46 @@
+/* Config:                                                                    //indent:0 exp:0
+ * This test-input is intended to be checked using following configuration:   //indent:1 exp:1
+ *                                                                            //indent:1 exp:1
+ * arrayInitIndent = 2                                                        //indent:1 exp:1
+ * basicOffset = 2                                                            //indent:1 exp:1
+ * braceAdjustment = 2                                                        //indent:1 exp:1
+ * caseIndent = 2                                                             //indent:1 exp:1
+ * forceStrictCondition = false                                               //indent:1 exp:1
+ * lineWrappingIndentation = 4                                                //indent:1 exp:1
+ * tabWidth = 4                                                               //indent:1 exp:1
+ * throwsIndent = 4                                                           //indent:1 exp:1
+ *                                                                            //indent:1 exp:1
+*/                                                                            //indent:0 exp:0
+
+package com.puppycrawl.tools.checkstyle.checks.indentation.indentation; //indent:0 exp:0
+
+class InputIndentationChainedMethodCalls { //indent:0 exp:0
+
+  public InputIndentationChainedMethodCalls(Object... params) { //indent:2 exp:2
+  } //indent:2 exp:2
+
+  public String doNothing(String something, String... uselessParams) { //indent:2 exp:2
+    return something; //indent:4 exp:4
+  } //indent:2 exp:2
+
+  public InputIndentationChainedMethodCalls getInstance(String... uselessParams) { //indent:2 exp:2
+    return this; //indent:4 exp:4
+  } //indent:2 exp:2
+
+  public static void main(String[] args) { //indent:2 exp:2
+    new InputIndentationChainedMethodCalls().getInstance("string_one") //indent:4 exp:4
+    .doNothing("string_one".trim(), //indent:4 exp:8 warn
+               "string_two"); //indent:15 exp:>=8
+
+    int length = new InputIndentationChainedMethodCalls("param1", //indent:4 exp:4
+                                "param2").getInstance() //indent:32 exp:>=8
+    .doNothing("nothing") //indent:4 exp:>=8 warn
+    .length(); //indent:4 exp:>=8 warn
+
+    int length2 =  //indent:4 exp:4
+    new InputIndentationChainedMethodCalls("param1","param2") //indent:4 exp:>=8 warn
+        .getInstance() //indent:8 exp:8
+        .doNothing("nothing") //indent:8 exp:8
+        .length(); //indent:8 exp:8
+  } //indent:2 exp:2
+} //indent:0 exp:0

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationCustomAnnotation.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationCustomAnnotation.java
@@ -84,7 +84,7 @@ class InnerAnnotSingleLine { //indent:0 exp:0
         public String toString() { //indent:8 exp:8
             return //indent:12 exp:12
                     new String( //indent:20 exp:20
-                            new StringBuilder("Hello") //indent:28 exp:28
+                            new StringBuilder("Hello") //indent:28 exp:20,24 warn
                                     .append(",World")) //indent:36 exp:36
                     ; //indent:20 exp:20
         } //indent:8 exp:8

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationNewWithForceStrictCondition.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationNewWithForceStrictCondition.java
@@ -1,0 +1,27 @@
+/* Config:                                                                    //indent:0 exp:0
+ * This test-input is intended to be checked using following configuration:   //indent:1 exp:1
+ *                                                                            //indent:1 exp:1
+ * arrayInitIndent = 4                                                        //indent:1 exp:1
+ * basicOffset = 4                                                            //indent:1 exp:1
+ * braceAdjustment = 0                                                        //indent:1 exp:1
+ * caseIndent = 4                                                             //indent:1 exp:1
+ * forceStrictCondition = true                                                //indent:1 exp:1
+ * lineWrappingIndentation = 8                                                //indent:1 exp:1
+ * tabWidth = 4                                                               //indent:1 exp:1
+ * throwsIndent = 4                                                           //indent:1 exp:1
+ *                                                                            //indent:1 exp:1
+ */                                                                           //indent:1 exp:1
+package com.puppycrawl.tools.checkstyle.checks.indentation.indentation;       //indent:0 exp:0
+import java.io.BufferedReader; //indent:0 exp:0
+import java.io.IOException; //indent:0 exp:0
+import java.io.InputStreamReader; //indent:0 exp:0
+
+public class InputIndentationNewWithForceStrictCondition { //indent:0 exp:0
+    void test() throws IOException  { //indent:4 exp:4
+        BufferedReader bf =  //indent:8 exp:8
+                new BufferedReader(  //indent:16 exp:16
+                new InputStreamReader(System.in) {  //indent:16 exp:24 warn
+                    int a = 0; //indent:20 exp:28,32,36 warn
+                }); //indent:16 exp:24,28,32 warn
+    } //indent:4 exp:4
+}  //indent:0 exp:0

